File: samples/go/05wc/wc.go
package main

import (
    "fmt"
    "os"
    "bufio"
)


func main() {
    if len(os.Args) > 2 {
        fmt.Fprintf(os.Stderr, "Error: Usage %s [<file>]\n", os.Args[0]);
        os.Exit(1);
    }

    var file = os.Stdin;

    if len(os.Args) == 2 {
        _file, err := os.Open(os.Args[1]);
        if err != nil {
            fmt.Fprintf(os.Stderr, "Error: Failed to open %s\n", os.Args[1]);
            os.Exit(2);
        }
        defer _file.Close();
        file = _file;
    }

    scanner := bufio.NewScanner(file);
    lines  := 0;
    for scanner.Scan() {
        lines += 1;
    }

    fmt.Printf("%d\n", lines);
}

File: samples/php/05wc/wc.php
<?php

if ($argc > 2) {
    fwrite(STDERR, sprintf("Error: Usage ./%s [<file>]\n", $argv[0]));
    exit(1);
}

$file = STDIN;
if ($argc == 2) {
    $file = fopen($argv[1], "r");
}
$lines = 0;

while (! feof($file)) {
    fgets($file);
    $lines += 1;
}

printf("%d\n", $lines);

?>
